<launch>
  <node
    pkg="turtlesim"
    type="turtlesim_node"
    name="turtlesim"
  />
  <node
    pkg="raven_img_recorder"
    type="test_turtlesim_random"
    name="velocity_publisher"
  />

  <node name="usb_cam0" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="mjpeg" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
    <param name="framerate" value="30"/>

  </node>
  <node
    pkg="image_view"
    type="image_view"
    name="image_view0"
    args="image:=/usb_cam0/image_raw"
  />
  <node
    pkg="rosbag"
    type="record"
    name="camera0_recorder"
    args="-j -O camera0_recorder.bag /usb_cam0/image_raw/compressed"
  />

  <node name="usb_cam1" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video1" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="mjpeg" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
    <param name="framerate" value="30"/>

  </node>
  <node
    pkg="image_view"
    type="image_view"
    name="image_view1"
    args="image:=/usb_cam1/image_raw"
  />
  <node
    pkg="rosbag"
    type="record"
    name="camera1_recorder"
    args="-j -O camera1_recorder.bag /usb_cam1/image_raw/compressed"
  />

  <node name="usb_cam2" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video2" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="mjpeg" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
    <param name="framerate" value="30"/>

  </node>
  <node
    pkg="image_view"
    type="image_view"
    name="image_view2"
    args="image:=/usb_cam2/image_raw"
  />
  <node
    pkg="rosbag"
    type="record"
    name="camera2_recorder"
    args="-j -O camera2_recorder.bag /usb_cam2/image_raw/compressed"
  />

<node name="usb_cam3" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video3" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="mjpeg" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
    <param name="framerate" value="30"/>

  </node>
  <node
    pkg="image_view"
    type="image_view"
    name="image_view3"
    args="image:=/usb_cam3/image_raw"
  />
  <node
    pkg="rosbag"
    type="record"
    name="camera3_recorder"
    args="-j -O camera3_recorder.bag /usb_cam3/image_raw/compressed"
  />

  <node
    pkg="rosbag"
    type="record"
    name="turtlesim_recorder"
    args="-j -O turtlesim_recorder.bag /turtle1/cmd_vel_throttle"
  />

  <node
    pkg="topic_tools"
    type="throttle"
    name="throttle"
    args="messages /turtle1/cmd_vel 100.0"
  />
</launch>
